code/modules/mob/living/basic/farm_animals/mushroom.dm

```diff
@@ -0,0 +1,185 @@
+	health = maxHealth
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_WALKING_MUSHROOM, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+/datum/ai_controller/basic_controller/mushroom
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/mushroom(),
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree/mushroom,
+		/datum/ai_planning_subtree/find_and_hunt_target/mushroom_food,
+	)
+
+
+/datum/targetting_datum/basic/mushroom
+	stat_attack = DEAD
+
+///we only attacked another mushrooms
+/datum/targetting_datum/basic/mushroom/faction_check(mob/living/living_mob, mob/living/the_target)
+	return !living_mob.faction_check_mob(the_target, exact_match = check_factions_exactly)
+
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/mushroom
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/mushroom
+
+/datum/ai_behavior/basic_melee_attack/mushroom
+	action_cooldown = 2 SECONDS
+
+/datum/ai_planning_subtree/find_and_hunt_target/mushroom_food
+	target_key = BB_LOW_PRIORITY_HUNTING_TARGET
+	hunting_behavior = /datum/ai_behavior/hunt_target/unarmed_attack_target/mushroom_food
+	hunt_targets = list(/obj/item/food/grown/mushroom)
+	hunt_range = 6
+
+
+/datum/ai_behavior/hunt_target/unarmed_attack_target/mushroom_food
+	hunt_cooldown = 15 SECONDS
+	always_reset_target = TRUE
+
+/mob/living/basic/mushroom/UnarmedAttack(atom/attack_target, proximity_flag, list/modifiers)
+	. = ..()
+	if(!.)
+		return
+
+	if(!proximity_flag)
+		return
+
+	if(istype(attack_target, /obj/item/food/grown/mushroom))
+		recover(attack_target)
+		return TRUE
+
+/mob/living/basic/mushroom/melee_attack(atom/target, list/modifiers)
+	. = ..()
+
+	if(!.)
+		return
+
+	if(!istype(target, /mob/living/basic/mushroom))
+		return
+	var/mob/living/basic/mushroom/victim = target
+	if(victim.stat != DEAD)
+		return
+	if(victim.faint_ticker < 2)
+		victim.faint_ticker++
+		src.visible_message(span_notice("[src] chews a bit on [victim]."))
+		return
+
+	consume_mushroom(victim)
+
+/mob/living/basic/mushroom/proc/consume_mushroom(mob/living/basic/mushroom/consumed)
+	src.visible_message(span_warning("[src] devours [consumed]!"))
+	var/level_gain = (consumed.powerlevel - powerlevel)
+	if(level_gain >= 0 && !ckey && !consumed.bruised)//Player shrooms can't level up to become robust gods.
+		consumed.LevelUp(level_gain)
+	adjustBruteLoss(-consumed.maxHealth)
+	qdel(consumed)
+
+/mob/living/basic/mushroom/revive(full_heal_flags = NONE, excess_healing = 0, force_grab_ghost = FALSE)
+	. = ..()
+	if(!.)
+		return
+
+	icon_state = "mushroom_color"
+	UpdateMushroomCap()
+
+/mob/living/basic/mushroom/death(gibbed)
+	. = ..()
+	UpdateMushroomCap()
+
+/mob/living/basic/mushroom/proc/UpdateMushroomCap()
```

Comment:
new code should be in `snake_case`, so you would just change this proc name (and everything that references it) as:

```suggestion
/mob/living/basic/mushroom/proc/update_mushroom_cap()
```

Just something that can be easily and readily done since we're cleaning up code.